packages/jaeger-ui/src/components/common/UIFindInput.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as React from 'react';
+import { Input } from 'antd';
+import _debounce from 'lodash/debounce';
+import _filter from 'lodash/filter';
+import _get from 'lodash/get';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import queryString from 'query-string';
+
+import type { Location, RouterHistory } from 'react-router-dom';
+
+import filterSpans from '../../utils/filter-spans';
+import updateUiFind from '../../utils/update-ui-find';
+
+import type { ReduxState } from '../../types/index';
+import type { Span } from '../../types/trace';
+
+type PropsType = {
+  forwardedRef?: { current: Input | null },
+  inputProps: Object,
+  history: RouterHistory,
+  location: Location,
+  spansArray?: Span[][],
+  uiFind?: string,
+};
+
+type StateType = {
+  ownInputValue: ?string,
+};
+
+export class UnconnectedUiFindInput extends React.PureComponent<PropsType, StateType> {
+  static defaultProps = {
+    forwardedRef: null,
+    inputProps: {},
+    spansArray: [],
+    uiFind: null,
+  };
+
+  state = {
+    ownInputValue: null,
+  };
+
+  handleInputBlur = () => {
+    this.updateUiFindQueryParam.flush();
+    this.setState({ ownInputValue: null });
+  };
+
+  handleInputChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    const { value } = evt.target;
+    this.updateUiFindQueryParam(value);
+    this.setState({ ownInputValue: value });
+  };
+
+  updateUiFindQueryParam = _debounce((uiFind: ?string) => {
+    const { history, location } = this.props;
+    updateUiFind({
+      location,
+      history,
+      uiFind,
+    });
+  }, 250);
+
+  render() {
+    const inputValue =
+      typeof this.state.ownInputValue === 'string' ? this.state.ownInputValue : this.props.uiFind;
+    const suffix: number | null =
+      this.props.uiFind && this.props.spansArray
+        ? _filter(this.props.spansArray, spans =>
+            _get(filterSpans(this.props.uiFind /* inputValue */ || '', spans), 'size', 0)
```

Comment:
@tiffon I'm interested in your perspective here. if I use `this.props.uiFind`, the count will reflect the current highlighted nodes, and I can `_memoize` `filterSpans` to make the renders caused by typing even faster.
I could instead use `inputValue` (as I have commented out). even on the diff graph I used in [this screen capture](https://github.com/jaegertracing/jaeger-ui/pull/310#issuecomment-466113601) it was quick enough to filter through the spans on keystroke. The main downside is that the count will update before the graph and that may be a bit wonky
A compromise could be to only show the count when `this.state.ownInputValue === this.props.uiFind || !this.state.ownInputValue` and otherwise show some indication of computing.

Reply:
antd input does support react elements as a suffix, so a loading indicator of sorts would work.